### PR TITLE
Use TrackerWithSnapshot for BaseUdpEndpoint metrics

### PR DESCRIPTION
**Pull request by reddvl1980**

### CODE REVIEW

File: utils/Trackers.h [hunk 1]
```diff
@@ -133,6 +133,26 @@ class RateTracker
     uint64_t _prevTimestamp;
 };
 
+template <size_t NUM_BUCKETS, uint64_t BUCKET_SIZE_MS, uint64_t SNAPSHOT_DELTA_MS>
+struct TrackerWithSnapshot
+{
+    TrackerWithSnapshot() : tracker(BUCKET_SIZE_MS), _lastSnapshotUpdatedAt(0) { snapshot.store(0.0); }
+    utils::RateTracker<NUM_BUCKETS> tracker;
+    std::atomic<double> snapshot;
```
Comment (olofkallander) [inline utils/Trackers.h:141]: there is a typedef for this std::atomic_double

File: transport/BaseUdpEndpoint.h [hunk 2]
```diff
@@ -71,6 +71,18 @@ class BaseUdpEndpoint : public Endpoint, RtcePoll::IEventListener
         memory::UniquePacket packet;
     };
 
+    struct RateMetrics
+    {
+        utils::TrackerWithSnapshot<10, utils::Time::ms * 100, utils::Time::sec> receiveTracker;
+        utils::TrackerWithSnapshot<10, utils::Time::ms * 100, utils::Time::sec> sendTracker;
+        EndpointMetrics toEndpointMetrics(size_t size) const
+        {
+            return EndpointMetrics(size,
```
Comment (olofkallander) [inline transport/BaseUdpEndpoint.h:80]: should we rename it queueSize to be more specific what is stored in EndpointMetrics.

**Comment by olofkallander** [comment]
I like it. One rename perhaps.

File: utils/Trackers.h (hunk 1 above)
Comment (reddvl1980) [reply utils/Trackers.h:141]: Are you sure?

Comment (reddvl1980) [reply utils/Trackers.h:141]: I mean atomic<double> is not fully implemented, and I can't find atomic_double macro.

File: transport/BaseUdpEndpoint.h (hunk 2 above)
Comment (reddvl1980) [reply transport/BaseUdpEndpoint.h:80]: Done.

**Pull request merged**